common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -245,12 +245,12 @@ public CollationIdentifier identifier() {
       PROVIDER_ICU,
       Collator.getInstance(ULocale.ROOT),
       "153.120.0.0",
-      true,
+      false,
```

Comment:
not for reviewers: this is the place we say UNICODE does not supportBinaryEquality